include/TFEL/Material/Eshelby.hxx

```diff
@@ -0,0 +1,108 @@
+   * and in Eshelby 1957 for other cases.
+   * \tparam    real: underlying type
+   * param 1 : nu
+   * param 2 : semi-axis of ellipsoid in direction 1
+   * param 3 :semi-axis of ellipsoid in direction 2
+   * param 4 : semi-axis of ellipsoid in direction 3
+   */
+     template <typename real>
+  TFEL_HOST_DEVICE const tfel::math::st2tost2<3u,real> EshelbyTensor(
+      const real&,
+      const real&,
+      const real&,
+      const real&);
+      
+  /*!
+   * This function builds the localisation tensor for an ellipsoid whose semi-axis lengths are a,b,c,
+   * whose elasticity tensor is C_i, 
+   * and embedded in an isotropic matrix (young,nu).
+   * The localisation tensor A is defined as follows : eps=A:eps0 where eps0 is the uniform strain tensor imposed at infinity,
+   * and eps is the strain tensor solution of Eshelby problem for the ellipsoid.
+   * The ellipsoid also has a specific orientation given by the vectors n_a, n_b.
+   * \returns an object of type st2tost2, which is the fourth-order localisation tensor A.
+   * \tparam    real: underlying type
+   * param 1,2 : young,nu : parameters of the isotropic matrix
+   * param 3 : st2tost2<3u,real> C_i : elastic tensor of the ellipsoid (may be anisotropic).
+   * param 4 : n_a : vector<real>,  direction of the principal axis whose length is a
+   * param 5 : a : length of semi-axis relative to the direction n_a
+   * param 6: n_b : vector<real>,  direction of the principal axis whose length is b
+   * param 7 : b : length of semi-axis relative to the direction n_b
+   * param 8 : c : length of the remaining semi-axis
+   */
+     template <typename real, typename StressType>
+  TFEL_HOST_DEVICE const tfel::math::st2tost2<3u,real> GeneralEllipsoidLocalisationTensor(
+      const StressType&,
+      const real&,
+      const tfel::math::st2tost2<3u, StressType>&,
+      const std::vector<real>&,
+      const real&,
+      const std::vector<real>&,
+      const real&,
+      const real&);
+      
+     /*
+      * The same function as GeneralEllipsoidLocalisationTensor for the case of an axisymetric ellipsoid
+      * The geometrical parameters are now n_a, the axis of the ellipsoid, and the aspect ratio e=a/b
+      */
+     template <typename real, typename StressType>
+  TFEL_HOST_DEVICE const tfel::math::st2tost2<3u,real> AxisymEllipsoidLocalisationTensor(
+      const StressType&,
+      const real&,
+      const tfel::math::st2tost2<3u, StressType>&,
+      const std::vector<real>&,
+      const real&);
+      
+      /*
+      * The same function as EllipsoidLocalisationTensor for the case of a sphere
+      */
+    template <typename real, typename StressType>
+  TFEL_HOST_DEVICE const tfel::math::st2tost2<3u,real> SphereLocalisationTensor(
+      const StressType&,
+      const real&,
+      const tfel::math::st2tost2<3u, StressType>&,
+      const std::vector<real>&,
+      const real&,
+      const std::vector<real>&,
```

Comment:
`std::vector` uses dynamic memory allocation. Use fixed size arrays like `std::array`, `tfel::math::fsarray` or `tfel::math::tvector<3u, real>`

In this case, `tfel::math::tvector<3u, real>` is a better choice